src/main/java/frc/robot/vision/limelights/LimelightFilterer.java

```diff
@@ -0,0 +1,119 @@
+package frc.robot.vision.limelights;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.poseestimator.PoseArrayEntryValue;
+import frc.robot.poseestimator.PoseEstimationMath;
+import frc.robot.poseestimator.observations.VisionObservation;
+import frc.robot.subsystems.GBSubsystem;
+import frc.utils.Conversions;
+import org.littletonrobotics.junction.Logger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public class LimelightFilterer extends GBSubsystem implements ILimelightFilterer {
+
+	private final MultiLimelights multiLimelights;
+	private final LimelightFiltererConfig config;
+	private double lastSuccessfulObservationTime;
+	private Function<Double, Pose2d> getEstimatedPoseAtTimestamp;
+
+	public LimelightFilterer(LimelightFiltererConfig config, MultiLimelights multiLimelights) {
```

Comment:
Is there any reason to not inject an `LimelightFiltersTolerances` to the limelightFilterer? This allows creating multiple filterers with different constants in parallel and checking which one is the best.

Reply:
I will just put it in the configs